crawl-ref/source/l-travel.cc

```diff
@@ -112,6 +112,28 @@ LUAFN(l_waypoint_delta)
     return 2;
 }
 
+/*** Set a numbered waypoint.
+ * Uses player-centered coordinates.
+ * @tparam int waynum
+ * @tparam int x
+ * @tparam int y
+ * @function set_waypoint
+ */
+LUAFN(l_set_waypoint)
+{
+    int waynum = luaL_safe_checkint(ls, 1);
+    if (waynum < 0 || waynum >= TRAVEL_WAYPOINT_COUNT)
+        return luaL_error(ls, "Bad waypoint number: %d", waynum);
+    coord_def s;
+    s.x = luaL_safe_checkint(ls, 2);
+    s.y = luaL_safe_checkint(ls, 3);
+    const coord_def p = player2grid(s);
+    if (!in_bounds(p))
+        return luaL_error(ls, "Coordinates out of bounds: (%d, %d)", p.x, p.y);
```

Comment:
I like the idea of more useful error messages, but this makes it even easier to figure out absolute coordinates, right?

With this in my rcfile:

```
{
  function setwp()
    travel.set_waypoint(0, 200, 200)
  end
}
```

I can easily discover from the error message that my current absolute position is (59, 50):

```
Lua error: ...amalloy/src/dcss/crawl-ref/source/dat/dlua/macro.lua:31: [string
 "/home/amalloy/.crawl/init...."]:2: Coordinates out of bounds:
 (259, 250)
```

Reply:
Whoops, didn't realize lua errors were visible but that makes a lot of sense upon further thought :) I'll change that to player-centered coords.